book/src/values.md

```diff
@@ -0,0 +1,30 @@
+# POD value types
+From the frontend perspective, POD values may be one of three[^type] types:
+- `Integer`
+- `String`
+- `MerkleTree`
+
+From the backend perspective, however, these types will all be encoded as a fixed number of field elements, the number being chosen so as to accommodate the `Integer` type as well as hashes to represent the `String` and `MerkleTree` types with the appropriate level of security.
+
+In the case of the Plonky2 backend with 100 bits of security, all of these types are represented as 4 field elements, the output of the Poseidon hash function used there being
+
+$$\texttt{HashOut<GoldilocksField>}\simeq\texttt{[GoldilocksField; 4]}.$$
+
+## `Integer`
+In the frontend, this type is none other than `u64`[^i64]. In the backend, it will be appropriately embedded into the codomain of the canonical hash function.
+
+In the case of the Plonky2 backend, this is done by decomposing such an integer $x$ as
+$$x = x_0 + x_1 \cdot 2^{32}$$
+with $0 \leq x_0, x_1 < 2^{32}$ and representing it as
+$$\texttt{map}\ \iota\ [x_0, x_1, 0, 0],$$
+where $\iota:\mathbb{N}\rightarrow\texttt{GoldilocksField}$ is the canonical projection.
+
+## `String`
+In the frontend, this type corresponds to the usual `String`. In the backend, the string will be mapped to a sequence of field elements[^String] and hashed with the hash function employed there, thus being represented by its hash.
+
+## `MerkleTree`
+In the front end, this type encapsulates a sparse Merkle tree. In the backend, this will be represented by the root hash of the tree, which will be of the type of the output of the hash function employed.
```

Comment:
is it or maybe the reverse? like in the frontend the `MerkleTree` is just the root, and in the backend it is the whole merkletree structure

Reply:
I think it can be seen both ways.
If you think about the `Value`, in the backend a `MerkleTree` `Value` is just the root.
In the frontend when we have a `MerkleTree` Value we can attach metadata (the entire tree, which is actually implemented beyond the frontend, probably the backend)